src/api/firebase.js

```diff
@@ -57,10 +57,11 @@ export function useShoppingLists(userId, userEmail) {
  * @param {string | null} listPath
  * @see https://firebase.google.com/docs/firestore/query-data/listen
  */
+// Start with an empty array for our data.
+/** @type {import('firebase/firestore').DocumentData[]} */
+const initialState = [];
```

Comment:
Since `initialState` is an array, it changes its value on every render, and since it's added as a dependency to the `useEffect`, it causes the component to re-render when it's changed. It should be good now!

<img width="1118" alt="Screenshot 2024-04-01 at 2 34 48 PM" src="https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/1530227/33e1603f-de73-4b34-8542-c2f903ba91a2">